recipes/ncbi-cxx-toolkit-public/26/conanfile.py

```diff
@@ -0,0 +1,1438 @@
+        return cmake
+
+#----------------------------------------------------------------------------
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 17)
+        if self.settings.os not in ["Linux", "Macos", "Windows"]:   
+            raise ConanInvalidConfiguration("This operating system is not supported")
+        if self.settings.compiler == "Visual Studio" and str(self.settings.compiler.version) < "15":
+            raise ConanInvalidConfiguration("This version of Visual Studio is not supported")
+        if self.settings.compiler == "Visual Studio" and self.options.shared and "MT" in self.settings.compiler.runtime:
+            raise ConanInvalidConfiguration("This configuration is not supported")
+        if self.settings.compiler == "gcc" and str(self.settings.compiler.version) < "7":
+            raise ConanInvalidConfiguration("This version of GCC is not supported")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+#----------------------------------------------------------------------------
+    def requirements(self):
+        NCBIreqs = self._get_RequiresMapKeys()
+        for req in NCBIreqs:
+            pkg = self._translate_ReqKey(req)
+            if pkg is not None:
+                self.requires(pkg)
+
+#----------------------------------------------------------------------------
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root = True)
+
+#----------------------------------------------------------------------------
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.configure(source_folder=self._source_subfolder)
+# Visual Studio sometimes runs "out of heap space"
+        if self.settings.compiler == "Visual Studio":
+            self.run("cmake --build . %s -j 1" % cmake.build_config)
+        else:
+            cmake.build()
+
+#----------------------------------------------------------------------------
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+
+#----------------------------------------------------------------------------
+    def package_info(self):
+        if self.settings.os == "Windows":
+            self.cpp_info.components["ORIGLIBS"].system_libs = ["ws2_32", "dbghelp"]
+            self.cpp_info.components["NETWORKLIBS"].system_libs = [""]
+        elif self.settings.os == "Linux":
+            self.cpp_info.components["ORIGLIBS"].system_libs = ["dl", "rt", "m", "pthread"]
+            self.cpp_info.components["NETWORKLIBS"].system_libs = ["resolv"]
+        elif self.settings.os == "Macos":
+            self.cpp_info.components["ORIGLIBS"].system_libs = ["dl", "c", "m", "pthread"]
+            self.cpp_info.components["NETWORKLIBS"].system_libs = ["resolv"]
```

Comment:
Is it a trick to easily set the same system libs for several other components? I yes, I would avoid to use a "fake component", because components have side effects (they automatically generate CMake imported targets and pkgconfig files in few conan generators, so it can be confusing and lead to conflicts with other recipes if component names are too generic).

And you likely want `self.cpp_info.components["NETWORKLIBS"].system_libs = []` instead of `self.cpp_info.components["NETWORKLIBS"].system_libs = [""]`.